### PR TITLE
DEVPROD-3655 Add phase notifications to py-tpcc

### PR DESCRIPTION
**Pull request by SimonEismann**
This PR adds callbacks to the script added in [DEVPROD-3527](https://jira.mongodb.org/browse/DEVPROD-3527) to py-tpcc.

Patch:
https://spruce.mongodb.com/version/65a4f86e0305b9b4eb0d3586

### CODE REVIEW

File: pytpcc/tpcc.py [hunk 1]
```diff
@@ -53,6 +54,24 @@
     '%(asctime)s [%(funcName)s:%(lineno)03d] %(levelname)-5s: %(message)s'))
 logging.getLogger('').addHandler(console)
 
+NOTIFY_PHASE_START_PATH = '/data/workdir/src/flamegraph/notify_phase_start.py'
```
Comment (johndaniels) [inline pytpcc/tpcc.py:57]: Why is the notify command in the flamegraph repo?

File: pytpcc/tpcc.py [hunk 2]
```diff
@@ -53,6 +54,24 @@
...
+NOTIFY_PHASE_END_PATH = '/data/workdir/src/flamegraph/notify_phase_start.py'
+
+## ==============================================
+## noftifyDsiOfPhaseStart
+## ==============================================
+def noftifyDsiOfPhaseStart(phasename):
+    if os.path.isfile(NOTIFY_PHASE_START_PATH):
+        call(["python3", NOTIFY_PHASE_START_PATH, phasename])
```
Comment (johndaniels) [inline pytpcc/tpcc.py:65]: I think https://docs.python.org/3/library/subprocess.html#subprocess.run is the recommended way to run subprocess now. `call` is listed as an older API. Can we use run? Do we have Python newer than 3.5

Relatedly, do we want to check the status code of this and throw an exception? Run has a "check" parameter that can be used for that.

**Review by johndaniels — CHANGES_REQUESTED** [review]

File: pytpcc/tpcc.py (hunk 1 above)
Comment (SimonEismann) [reply pytpcc/tpcc.py:57]: The DSI repo would be the natural choice here, but it is not rsynced over to the workload client. The flamegraph repo is the most profiling related repo I could find and is a repo where the script is "out of the way".

From [this PR](https://github.com/mongodb-forks/flamegraph/pull/1):

### Why are the scripts in this repo?
These scripts would semantically make the most sense in the DSI repo. However, when running DSI locally, a) the DSI folder is not rsynced to the workload client, and b) the DSI folder on the local machine is in the directory above the work directory, while on Evergreen it's located in src/dsi. See [this Slack discussion](https://mongodb.slack.com/archives/C044WBW0189/p1704728988752149) for more details.

File: pytpcc/tpcc.py (hunk 2 above)
Comment (SimonEismann) [reply pytpcc/tpcc.py]: I've swapped to `subprocess.run` and checked for the error code. It's quite neat how `subprocess.run` returns a human-readable object that contains all relevant information:
```
CompletedProcess(args=['ls', '-l', '/dev/null'], returncode=0,
stdout=b'crw-rw-rw- 1 root root 1, 3 Jan 23 16:23 /dev/null\n', stderr=b'')
```
Patch for the new version:
https://spruce.mongodb.com/version/65a7b01e7742ae630812f6ec

**Review by johndaniels — APPROVED** [review]

**Pull request merged**